packages/riipen-ui/src/components/checkbox.test.jsx

```diff
@@ -0,0 +1,227 @@
+          .find("Checkbox")
+          .childAt(0)
+          .childAt(0)
+          .childAt(2)
+          .hasClass(color)
+      ).toEqual(true);
+    });
+
+    it("gives an error when given an invalid color", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+
+      mount(<Checkbox color="pink" readOnly />);
+
+      expect(errors).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("disabled prop", () => {
+    it("sets disabled with given disabled value", () => {
+      const disabled = true;
+
+      const wrapper = mount(<Checkbox disabled={disabled} readOnly />);
+
+      expect(
+        wrapper
+          .find("Checkbox")
+          .find("input")
+          .props().disabled
+      ).toEqual(disabled);
+    });
+
+    it("applies disabled class name when disabled is true", () => {
+      const disabled = true;
+
+      const wrapper = mount(<Checkbox disabled={disabled} readOnly />);
+
+      expect(
+        wrapper
+          .find("Checkbox")
+          .find(".checkmark")
+          .hasClass("disabled")
+      ).toEqual(true);
+    });
+  });
+
+  describe("error prop", () => {
+    it("displays given error", () => {
+      const error = <span>error!</span>;
+
+      const wrapper = mount(<Checkbox error={error} readOnly />);
+
+      expect(wrapper.childAt(0).contains(error)).toEqual(true);
+    });
+  });
+
+  describe("label prop", () => {
+    it("displays given label in the label element", () => {
+      const label = "this is a label";
+
+      const wrapper = mount(<Checkbox label={label} readOnly />);
+
+      expect(wrapper.find("label").contains(label)).toEqual(true);
+    });
+  });
+
+  describe("required prop", () => {
+    it("displays an asterisk at the end of the label when required is true", () => {
+      const label = <span>this is a label</span>;
+      const required = true;
+
+      const wrapper = mount(
+        <Checkbox required={required} label={label} readOnly />
+      );
+
+      expect(
+        wrapper
+          .find("Checkbox")
+          .find("label")
+          .text()
+          .substr(-1)
+      ).toEqual("*");
+    });
+
+    it("does not display an asterisk when required is false", () => {
+      const label = "this is a label";
+      const required = false;
+
+      const wrapper = mount(
+        <Checkbox required={required} label={label} readOnly />
+      );
+
+      expect(
+        wrapper
+          .find("Checkbox")
+          .find("label")
+          .text()
+          .substr(-1)
+      ).not.toEqual("*");
+    });
+  });
+
+  describe("warning prop", () => {
+    it("displays given warning", () => {
+      const warning = <span>warning!</span>;
+
+      const wrapper = mount(<Checkbox warning={warning} readOnly />);
+
+      expect(wrapper.childAt(0).contains(warning)).toEqual(true);
+    });
+  });
+
+  describe("focusVisible state", () => {
+    it("does not apply focusVisible class when component is rendered", () => {
```

Comment:
Is this test necessary? I'm wondering if we only need to test after events occur.

Reply:
Ya I think we can safely remove this. Imo makes a good point.